test/integration/Deposit.t.sol

```diff
@@ -94,9 +94,14 @@ contract DepositTest is BaseTest {
         assertEq(vault.maxMint(self), 0);
         centrifugeChain.updateMember(vault.poolId(), vault.trancheId(), self, type(uint64).max);
 
+        vm.assume(randomUser != self);
         // deposit 50% of the amount
-        vault.deposit(amount / 2, self); // mint half the amount
+        vm.startPrank(randomUser); // try to claim deposit on behalf of user and set the wrong user as receiver
+        vm.expectRevert(bytes("LiquidityPool/not-owner-or-endorsed"));
+        vault.deposit(amount / 2, randomUser, self);
+        vm.stopPrank();
 
+        vault.deposit(amount / 2, self, self); // deposit half the amount
```

Comment:
I will put the comment here but it applies to all occurrences of `vault.deposit`. You replaced "old" signature with the "new" one everywhere. Doing so, there is no test for the `deposit(uint256 assets, address receiver)` . I can break it and there is nothing that will catch it.
I literally hard-coded the return value and nothing broke for me:
```
  function deposit(uint256 assets, address receiver) external returns (uint256 shares) {
        return 9;
   }
```

Of course this won't be needed, if you decide to get rid of the "old" method.

Reply:
ERC7540 specifies (intentionally, to support the old interface for claiming) that both overloaded deposit methods should be included. So we shouldn't remove one, but I agree we should definitely add tests for both versions!